src/libstd/thread/mod.rs

```diff
@@ -367,6 +367,26 @@ impl Builder {
 /// want to specify the stack size or the name of the thread, use this API
 /// instead.
 ///
+/// As you can see in the signature of `spawn` there are two constraints on
+/// both the closure given to `spawn` and its return value, let's explain them:
+///
+/// - The `'static` constraint means that the closure and its return value
+///   must have a lifetime of the whole program execution. The reason for this
+///   is that threads can `detach` and outlive the lifetime they have been
+///   created in.  
```

Comment:
They are actually linked.

The value a thread returns must outlive the closure, but the fact that the thread can detach implies that the closure must be `'static`, since the value must live at least as long as the closure it must then also be `'static`